### PR TITLE
Fix Multiple issues with `_AsyncSingleShotTimer`

### PR DESCRIPTION
**Pull request by th3w1zard1**
This PR serves to fix the issues we had with 1.10.0, but currently will only highlight the issue.

I removed the check for `timer.is_running()` which shouldn't fail the test, yet it does.  We put a band-aid on 1.10.1, but we should still fix this.

Check the `reset()` function in `start_transition_timer`.  The `ValueError()` is never raised! What on earth?

### CODE REVIEW

**Comment by th3w1zard1** [comment]
I also ran into this problem addressed in https://github.com/basnijholt/adaptive-lighting/pull/529#issuecomment-1500875200

File: custom_components/adaptive_lighting/switch.py [hunk 1]
```diff
@@ -1692,7 +1692,7 @@ def start_transition_timer(self, light: str) -> None:
         )
 
         async def reset():
-            ValueError("TEST")
```
Comment (th3w1zard1) [inline custom_components/adaptive_lighting/switch.py:1695]: Looks like I forgot to remove this for 1.11.0. My bad!

File: custom_components/adaptive_lighting/switch.py [hunk 2]
```diff
@@ -1692,7 +1692,7 @@ def start_transition_timer(self, light: str) -> None:
...
+            raise ValueError("THIS SHOULD BE CALLED")
```
Comment (basnijholt) [inline custom_components/adaptive_lighting/switch.py:1695]: When running as a task this will probably not get noticed.

Comment (basnijholt) [reply custom_components/adaptive_lighting/switch.py:1695]: Only when later calling `task.result()` one will see the error.

**Pull request closed without merging**